src/net/tls.cc

```diff
@@ -1358,23 +1360,25 @@ class session : public enable_lw_shared_from_this<session> {
             gnutls_transport_set_errno(*this, EAGAIN);
             return -1;
         }
-        if (_output_pending.failed()) {
-            // copy exception for on_internal_error
-            auto ex = _output_pending.get_exception();
-            _output_pending = make_exception_future<>(ex);
-            on_internal_error(seastar_logger, format("tls: session: vec_push: unhandled pending output error: {}", ex));
-        }
         try {
-            scattered_message<char> msg;
-            for (int i = 0; i < iovcnt; ++i) {
-                msg.append(std::string_view(reinterpret_cast<const char *>(iov[i].iov_base), iov[i].iov_len));
+            ssize_t n; // Set on the good path and unused on the bad path
+            bool output_pending_failed = _output_pending.failed();
+
+            if (!output_pending_failed) {
+                scattered_message<char> msg;
+                for (int i = 0; i < iovcnt; ++i) {
+                    msg.append(std::string_view(reinterpret_cast<const char *>(iov[i].iov_base), iov[i].iov_len));
+                }
+                n = msg.size();
+                _output_pending = _out.put(std::move(msg).release());
             }
-            auto n = msg.size();
-            _output_pending = _out.put(std::move(msg).release());
             if (_output_pending.failed()) {
+                auto ex = _output_pending.get_exception();
                 // exception is copied back into _output_pending
                 // by the catch handlers below
-                std::rethrow_exception(_output_pending.get_exception());
+                auto level = output_pending_failed ? log_level::warn : log_level::debug;
+                seastar_logger.log(level, "tls: vec_push: {}output failed: {}", output_pending_failed ? "pending " : "", ex);
```

Comment:
Why the logging? To try to find a deeper issue in test runs? If so should be marked as temporary. If not it should not be here. We've established that reaching vec_push with _output_pending in error state (or any state except ready waited for) is an error, i.e. some oversight in the state machine of our tls.

Reply:
At the very least this logging should only be done in the actual real error case, i.e. we came in with a pending error.